### PR TITLE
feat: test-crawl with diff

### PR DESCRIPTION
**Pull request by rjsparks**
Ported from previous tzaware-obe branch

### CODE REVIEW

File: bin/test-crawl [hunk 1]
```diff
@@ -244,12 +246,104 @@ def get_referrers(url):
...
+    #open('a.html','wb').write(a)
+    #open('b.html','wb').write(b)
+    a = a.decode().splitlines()
+    b = b.decode().splitlines()
+    for group in difflib.SequenceMatcher(None,a,b).get_grouped_opcodes():
+        for tag, i1, i2, j1, j2 in group:
+            if tag == 'equal':
+                pass
+            elif tag == 'replace':
+                # see if the lines have next-day date matches
```
Comment (jennifer-richards) [inline bin/test-crawl:286]: Is this just to allow for the test run crossing over a day boundary?

**Review by jennifer-richards — APPROVED** [review]
Looks reasonable.  A couple of questions inline.

File: bin/test-crawl [hunk 2]
```diff
@@ -244,12 +246,104 @@ def get_referrers(url):
         ref_list.append(url)
     return ref_list
 
+nowtime = None
+nowstrn = None
+def nowstr():
+    global nowtime, nowstrn
+    if nowtime != time.time():
```
[the remark this answers is not in the document]
Comment (rjsparks) [reply bin/test-crawl:253]: looks to me to be improper optimization - I suspect the real condition that is interesting is whether the nowstrn computation would return a different string

File: bin/test-crawl [hunk 3]
```diff
@@ -244,12 +246,104 @@ def get_referrers(url):
...
+    return nowstrn
+
+b_exthost = re.sub(b'https?', b'', args.exthost.encode()) if args.exthost else None
+def normalize_for_diff(page):
+    # pages containing 'current time' can differ if they're fetched on different seconds:
+    #page = page.replace(nowstr(), b'00:00:00')
+    page = page.replace(b'https://', b'http://')
+    # regex replacements
+    page = re.sub(b'<!--.*?-->', b'', page)
+    page = re.sub(b' -- (Test|Development) Mode', b'', page)
```
[the remark this answers is not in the document]
Comment (rjsparks) [reply bin/test-crawl:265]: say more - where are you seeing what it is trying to abstract away?

File: bin/test-crawl (hunk 1 above)
Comment (rjsparks) [reply bin/test-crawl:286]: yes, though there's concern with it masking a date round difference in the long run. Let's keep it for now, but remember to try a run without it to see what differences it turns up.

File: bin/test-crawl (hunk 2 above)
[the remark this answers is not in the document]
Comment (rjsparks) [reply bin/test-crawl]: Seriously premature optimization, but whatever - I improved it some, but just building the string every time isn't even noticeable on a test run scale:
```
>>> timeit.timeit(setup='from django.utils import timezone',stmt='timezone.now().strftime("%H:%M:%S").encode()',number=100000)
0.16610741696786135
```

**Pull request merged**